### DISCUSSION TITLE
Hyperparameter tuning in torchdistill

### DISCUSSION THREAD

**Discussion by nighting0le01** [yoshitomo-matsubara/torchdistill#381] [Q&A]
question 1 : Hi @yoshitomo-matsubara how can I specify to using only a small subset of imagenet train for hyperparameter tuning ? I have the dataset locally downloaded. Can i specify it through yaml?

**Comment by yoshitomo-matsubara** [comment]
You can use [Subset class in torchvision](https://pytorch.org/docs/stable/data.html#torch.utils.data.Subset) and register the class or function to instantiate so that you can define it (as `imagenet_subset` is the name of registered dataset in the following example) in a yaml config

```python
from torch.utils.data import ImageFolder, Subset
from torchdistill.datasets.registry import register_dataset

@register_dataset
def imagenet_subset(indices, **image_folder_kwargs):
	org_dataset = ImageFolder(**image_folder_kwargs)
	sub_dataset = Subset(org_dataset, indices)
	return sub_dataset
```

Another option is randomly splitting your dataset like [this (dummy_train:dummy_val = 9:1) ](https://github.com/yoshitomo-matsubara/torchdistill/blob/v0.3.3/configs/sample/cifar10/ce/densenet_bc_k12_depth100-hyperparameter_tuning.yaml#L6-L37)
